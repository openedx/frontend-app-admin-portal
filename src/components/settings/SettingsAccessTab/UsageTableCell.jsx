import PropTypes from 'prop-types';

const UsageTableCell = ({ row }) => {
  const { usageCount, usageLimit } = row.original;
  return `${usageCount.toLocaleString()} of ${usageLimit.toLocaleString()}`;
};

UsageTableCell.propTypes = {
  row: PropTypes.shape({
    original: PropTypes.shape({
      usageCount: PropTypes.number,
      usageLimit: PropTypes.number,
    }),
  }).isRequired,
};

export default UsageTableCell;
