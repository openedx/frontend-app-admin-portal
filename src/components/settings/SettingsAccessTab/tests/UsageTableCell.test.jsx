import React from 'react';
import renderer from 'react-test-renderer';

import UsageTableCell from '../UsageTableCell';

describe('UsageTableCell', () => {
  it('renders correctly', () => {
    const props = {
      row: {
        original: {
          usageCount: 10,
          usageLimit: 100,
        },
      },
    };
    const tree = renderer
      .create(<UsageTableCell {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
