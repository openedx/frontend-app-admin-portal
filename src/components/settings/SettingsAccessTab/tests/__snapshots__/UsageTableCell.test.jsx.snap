// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageTableCell renders correctly 1`] = `"10 of 100"`;
